src/Bicep.Core/Semantics/NameBindingVisitor.cs

```diff
@@ -239,6 +239,16 @@ public override void VisitParameterDeclarationSyntax(ParameterDeclarationSyntax
             allowedFlags = FunctionFlags.Default;
         }
 
+        public override void VisitAssertDeclarationSyntax(AssertDeclarationSyntax syntax)
+        {
+            this.VisitNodes(syntax.LeadingNodes);
+            this.Visit(syntax.Keyword);
+            this.Visit(syntax.Name);
+            this.Visit(syntax.Assignment);
+            this.Visit(syntax.Value);
+            allowedFlags = FunctionFlags.Default;
+        }
+
```

Comment:
```suggestion
```

Since you're not setting `allowedFlags` prior to visiting the various components of the syntax node, you can just let this visitor fall back on the implementation in the base class (which is identical to lines 244-248).